rules_engine/admin.py

```diff
@@ -1,37 +1,183 @@
+"""
+.. _admin:
+
+django admin for the rules_engine app
+
+:module:    p_soc_auto.rules_engine.admin
+
+:copyright:
+
+    Copyright 2018 Provincial Health Service Authority
+    of British Columbia
+
+:contact:    serban.teodorescu@phsa.ca
+
+:updated:    sep. 18, 2018
+"""
 from django.contrib import admin
+from django.contrib.auth import get_user_model
+from django.contrib.contenttypes.models import ContentType
+
+from p_soc_auto_base.admin import BaseAdmin
 
 from .models import (
     TinDataForRuleDemos, IntervalRule, RuleApplies, ExpirationRule,
     NotificationEventForRuleDemo, RegexRule)
+from .forms import RuleAppliesForm
+
+
+class RulesEngineBaseAdmin(BaseAdmin, admin.ModelAdmin):
+    """
+    base admin class for this application
+    """
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        """
+        overload
+        admin.ModelAdmin.formfield_for_foreignkey(
+            self, db_field, request, **kwargs)
+        """
+        if db_field.name in ['created_by', 'updated_by', ]:
+            kwargs['queryset'] = get_user_model().objects.\
+                filter(username=request.user.username)
+            kwargs['initial'] = kwargs['queryset'].get()
+
+        if db_field.name in ['content_type', ]:
+            kwargs['queryset'] = ContentType.objects.\
+                filter(app_label__in=[
+                    'orion_integration', 'rules_engine', 'ssl_cert_tracker',
+                    'notifications'])
+
```

Comment:
why aren't you using else clause here instead of two ifs

Reply:
because the conditions are not if/else. if/else is binary choice. since we have more than 2 fields on the form, our choice is not binary